docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,181 @@
+:---------------------------- |:--------|:---------------------|:---------------------------------------------------------------------------------------
+Z01_MNAME_NOT_AUTHORITATIVE   | WARNING | ns                   | MNAME name server points to a non-authoritative name server ("{ns}").
+Z01_MNAME_NO_RESPONSE         | NOTICE  | ns                   | MNAME name server points to a non-responsive name server ("{ns}").
+Z01_MNAME_NOT_IN_ZONE         | INFO    | nsname               | MNAME name server {nsname} is not listed as NS record for the zone.
+Z01_MNAME_NOT_MASTER          | WARNING | ns_ip_list           | MNAME name server does not have the highest SERIAL. Returned from name servers "{ns_ip_list}".
+Z01_MNAME_IS_MASTER           | DEBUG   | ns_ip_list           | MNAME name server does appear to be master. Returned from name servers "{ns_ip_list}".
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create the following [DNS Queries][DNS Query]:
+   1. Query type SOA and query name *Child Zone* ("SOA Query")
+   2. Query type A and query name *Child Zone* ("A Query")
+   3. Query type AAAA and query name *Child Zone* ("AAAA Query")
+
+2. Create the following empty sets:
+   1. Name server IP address, name, SERIAL ("MNAME Nameservers")
+   2. Name server IP address, SERIAL ("SERIAL Nameservers")
+   3. Name server IP address ("MNAME Not Master")
+   4. Name server IP address ("MNAME Is Master")
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5] ("Name Server IP").
+
+4. For each name server IP in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, add the name server name from the MNAME field to the *MNAME Nameservers* set.
+   4. From the DNS response, add the SERIAL field and name server IP address to the *SERIAL Nameservers* set.
+   5. Go to next name server.
+
+5. If the set *MNAME Nameservers* is empty, then terminate these procedures.
```

Comment:
Either because, for some reason, (i) the set *Name Server IP* is empty (no name servers found), or (ii) no name server responded with a valid response (any conditions from step 4.1.2)